test/cluster/mzcompose.py

```diff
@@ -3520,3 +3520,67 @@ def check_frontier_is_not_stuck():
             )
         )
         check_frontier_is_not_stuck()
+
+
+def workflow_test_github_26215(c: Composition, parser: WorkflowArgumentParser) -> None:
+    """Regression test for #26215."""
+
+    def check_frontiers_advance():
+        c.sql(
+            """
+            BEGIN;
+            DECLARE c CURSOR FOR SUBSCRIBE (
+                SELECT write_frontier
+                FROM mz_internal.mz_frontiers
+                JOIN mz_sources ON (id = object_id)
+                WHERE name = 'lineitem'
+            );
+            FETCH 1 c;
+            FETCH 1 c;
+            FETCH 1 c;
+            FETCH 1 c;
+            FETCH 1 c;
+            ROLLBACK;
+            """
+        )
```

Comment:
If we regress, i.e. the frontiers don't advance, one of the `FETCH`es will hang forever. I haven't found a good way to configure a timeout that will cause a Python exception in that case (setting a timeout on `FETCH` will let the fetch succeed after the timeout, setting `statement_timeout` doesn't seem to do anything). Lmk if you know a way!

Reply:
Could the testdrive `set-sql-timeout` statement apply here?

E.g. if you add a 

```
c.testdrive("$ set-sql-timeout duration=1s", persistent=True)
```
before the `c.sql` call, does that timeout any hanging `FETCH` statements?

Reply:
I don't think so, since I'm not using testdrive here. I considered using testdrive, but that's annoying because it doesn't let me ignore the output of the `FETCH` statements, which consists of variable timestamps.

But we can regex-replace timestamps in testdrive. I might just have to bite the bullet and do that.

Reply:
Took me a while, but I managed to get the testdrive work!